engine/baml-lib/baml-core/src/ir/ir_helpers/to_baml_arg.rs

```diff
@@ -177,7 +177,9 @@ impl ArgCoercer {
             (FieldType::Enum(name), _) => match value {
                 BamlValue::String(s) => {
                     if let Ok(e) = ir.find_enum(name) {
-                        if e.walk_values().any(|v| v.item.elem.0 == *s) {
+                        if e.walk_values().any(|v| v.item.elem.0 == *s)
+                            || e.item.attributes.get("dynamic_type").is_some()
```

Comment:
Is this morally the same behavior that already exists for `Class` argument coersion in the case of dynamic types? https://github.com/BoundaryML/baml/pull/1343/files#diff-5fa0605b8093a86ff1b1b8e89032c02d46a052907bfa6dc38032c3ffc37aca02R238  I'm not sure, but it is. Just double-checking.

Reply:
Yes it's the same behavior. I think it makes sense, if you have a dynamic enum and add variants  to it at runtime, you would not be able to pass those variants to functions if the coercer doesn't allow it. We might want to restrict both classes and enums to only values added at runtime or in type builder blocks, but for now this does the job.